packages/koa-metrics/src/index.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable camelcase */
+import {StatsD} from 'hot-shots';
+import {Context} from 'koa';
+
+export interface Options {
+  prefix: string;
+  host: string;
+}
+
+export default function metrics({prefix, host}: Options) {
+  const statsdHost = host.split(':')[0];
+  const statsdPort = parseInt(host.split(':')[1], 10);
+
+  return async function statsdMiddleware(ctx: Context, next: Function) {
+    const requestStartTime = process.hrtime();
+    // @ts-ignore the types here are correct. TS complains about method
+    // potentially being undefined, but in reality, this is fine (see docs).
+    const client = new StatsD({
+      host: statsdHost,
+      port: statsdPort,
+      prefix,
+      globalTags: {
+        path: ctx.path,
+        request_method: ctx.request.method,
+      },
+    });
+
+    // allow later middleware to add metrics
+    ctx.state.statsd = client;
```

Comment:
It kind of bugs me when we do so much in each middleware. Could we break this up into the three/ four discrete things it is doing (set up client, timing for queuing, request details, maybe split between timing and status)

Reply:
I'd rather the users don't need to call `app.use` on 3/4 different middlewares just for metrics. Any idea what a suitable middleground might be? I'm thinking something like what we have now, but we can add a `skipDefaultMetrics` option?

Reply:
You can have one middleware that just composes a bunch of functions together

Reply:
To confirm, then, you're talking more about code style, rather than the middleware's functionality?

Reply:
Yes